src/errors.ts

```diff
@@ -32,8 +39,22 @@ export class BlockstackError extends Error {
   parameter?: string
 
   constructor(error: ErrorType) {
-    super(error.message)
-    this.message = error.message
+    let message = error.message
+    let bugMessage = `Error Code: ${error.code}`
+    let stack = null
```

Comment:
`Error.stack` is [weird](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack). Depending on the runtime, I think typically `this.stack` will already be set, and closer to the offending lines.

It might be better for this code to look something like 
```js
let stack = this.stack
if (!stack) { 
  try {
    throw new Error()
  } catch (e) {
     stack = e.stack
  }
}
```

Reply:
Unfortunately `this.stack` is inaccessible until I do `super(message)`, where `message` requires the stack :-/.

Reply:
You should be able to call `super(error.message)` or just `super()` at the beginning of the constructor and the later `this.message` reassignment should work as intended

Reply:
Good idea. Done :).###